### PR TITLE
Instrumentation

### PR DESCRIPTION
**Pull request by sgenie68**
Instrumentation support for vtune and gprof added

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -228,7 +228,7 @@ tidy_gui_variables()
 
 set(OPENCMISS_MPI ${OPENCMISS_MPI} CACHE STRING "Specify the MPI mnemonic.")
 set(OPENCMISS_BUILD_OWN_MPI NO CACHE BOOL "Build own MPI if system MPI is unavailable.")
-set(OPENCMISS_INSTRUMENTATION "none" CACHE STRING "Specify the instrumentation.")
+set(OC_INSTRUMENTATION "none" CACHE STRING "Specify the instrumentation.")
```
Comment (hsorby) [inline CMakeLists.txt]: The cache variable should be OPENCMISS_INSTRUMENTATION.  The variable used to manipulate this through the command line should be OC_INSTRUMENTATION.

**Comment by hsorby** [comment]
The instrumentation could be done as it's own configuration and not specified by changing the compiler names.

File: CMakeLists.txt (hunk 1 above)
Comment (chrispbradley) [reply CMakeLists.txt]: ?Isn't it the other way around? OPENCMISS_xxx on the command line? (I don't mind if it is OC_xxx on the command line or OPENCMISS_xxx - just that we should be consistent)

Comment (sgenie68) [reply CMakeLists.txt]: I think all other flags are OPENCMISS_xxx in the command line.

**Comment by chrispbradley** [comment]
> The instrumentation could be done as it's own configuration and not specified by changing the compiler names.

Do you mean to not change, say, cc to something like scorep-cc? If so then some of the instrumentation options require a different compiler (they wrap the compiler in order to insert there own code). Scorep is one of this (there are others)

File: CMakeLists.txt (hunk 1 above)
Comment (hsorby) [reply CMakeLists.txt]: All cache variables that may be manipulated in a GUI  that effect OpenCMISS specific parameters should be prefixed with OPENCMISS_.

Comment (sgenie68) [reply CMakeLists.txt]: It's not a problem to change the name, just want to be consistent as Chris said. I can use the same variable throughout the files, though

**Comment by sgenie68** [comment]
I renamed the variables to maintain consistency

**Pull request merged**